### PR TITLE
[INLONG-10879][SDK] Transform support TIMESTAMPADD() function

### LINKED ISSUE #10879: [Feature][SDK] Transform support TIMESTAMPADD() function
### Description

- parent issue #10796 

TIMESTAMPADD(unit,interval,datetime_expr) - Add integer expression intervals to the date or date time expression expr. The unit of the time interval is specified by the unit parameter, which should be one of the following values: FRAC_SECOND, SECOND, MINUTE, HOUR, DAY, WEEK, MONTH, QUARTER, or YEAR.

For example：
```sql
SELECT TIMESTAMPADD(MINUTE,1,'2003-01-02');
+---------------------------------------------------------+
|           TIMESTAMPADD(MINUTE,1,'2003-01-02')           |
+---------------------------------------------------------+
|                  2003-01-02 00:01:00                    |
+---------------------------------------------------------+
1 row in set (0.00 sec)
```

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Zkplo**
Fixes #10879 

### Motivation
Add one arithmetic function classe: TimestampAddFunction. Also, add the corresponding unit test codes.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TimestampAddFunction.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        String dateString = datetimeParser.parse(sourceData, rowIndex, context).toString();
+        return evalDate(dateString, interval, amount);
+    }
+
+    private String evalDate(String dateString, String interval, Long amount) {
+        DateTimeFormatter formatter = null;
+        LocalDateTime dateTime = null;
+        boolean hasTime = true;
+        if (dateString.indexOf(' ') != -1) {
+            formatter = DateTimeFormatter.ofPattern(DEFAULT_FORMAT_DATE_TIME);
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TimestampAddFunction.java:67]: DateTimeFormatter.ofPattern is a complex operation that affects performance, and needs to be constructed in advance as a member variable. Additionally, it needs to support thread safety, so it is recommended to use ThreadLocal.

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TimestampAddFunction.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            dateTime = LocalDateTime.parse(dateString, formatter);
+        } else {
+            formatter = DateTimeFormatter.ofPattern(DEFAULT_FORMAT_DATE);
+            dateTime = LocalDate.parse(dateString, formatter).atStartOfDay();
+            hasTime = false;
+        }
+
+        switch (interval.toUpperCase()) {
+            case "FRAC_SECOND":
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TimestampAddFunction.java:76]: why not support microsecond and millisecond?

Comment (Zkplo) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TimestampAddFunction.java:76]: "FRAC-SECOND" stands for milliseconds. No matter how high the accuracy is, other databases no longer support it. There is an error here that has been corrected.

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TimestampAddFunction.java (hunk 1 above)
Comment (aloyszhang) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TimestampAddFunction.java:67]: As @luchunliang said, we should not create a new `DateTimeFormatter` every time we call this function, this is time and memory costly.
It's better to initialize only one  `DateTimeFormatter` for every pattern.

Comment (Zkplo) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TimestampAddFunction.java]: Thank you, it has been modified. Please check.

**Review by aloyszhang — APPROVED** [review]

**Review by yfsn666 — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]

**Pull request merged**